src/deepsparse/transformers/eval_downstream.py

```diff
@@ -0,0 +1,283 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Example script for evaluating an ONNX model on a downstream dataset using the
+DeepSparse Engine or ONNXRuntime
+
+##########
+Command help:
+usage: eval_downstream.py [-h] [-d {squad,mnli,qqp,sst2}] [-c NUM_CORES]
+                          [-e {deepsparse,onnxruntime}]
+                          [--max-sequence-length MAX_SEQUENCE_LENGTH]
+                          [--max-samples MAX_SAMPLES]
+                          onnx_filepath
+
+Evaluate a BERT ONNX model on a downstream dataset
+
+positional arguments:
+  onnx_filepath         The full filepath of the ONNX model file or SparseZoo
+                        stub to the model
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -d {squad,mnli,qqp,sst2}, --dataset {squad,mnli,qqp,sst2}
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the eval on,
+                        defaults to all physical cores available on the system
+  -e {deepsparse,onnxruntime}, --engine {deepsparse,onnxruntime}
+                        Inference engine backend to run eval on. Choices are
+                        'deepsparse', 'onnxruntime'. Default is 'deepsparse'
+  --max-sequence-length MAX_SEQUENCE_LENGTH
+                        the max sequence length for model inputs. Default is
+                        384
+  --max-samples MAX_SAMPLES
+                        the max number of samples to evaluate. Default is None
+                        or all samples
+
+##########
+Example command for evaluating a sparse BERT QA model from sparsezoo:
+python eval_downstream.py \
+    zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none \
+    --dataset squad
+"""
+
+
+import argparse
+
+from tqdm.auto import tqdm
+
+from deepsparse.transformers import pipeline
+
+
+from datasets import load_dataset, load_metric  # isort: skip
+
+
+DEEPSPARSE_ENGINE = "deepsparse"
+ORT_ENGINE = "onnxruntime"
+
+SQUAD_KEY = "squad"
+MNLI_KEY = "mnli"
+QQP_KEY = "qqp"
+SST2_KEY = "sst2"
+SUPPORTED_DATASETS = [SQUAD_KEY, MNLI_KEY, QQP_KEY, SST2_KEY]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Evaluate a BERT ONNX model on a downstream dataset"
+    )
+    parser.add_argument(
+        "onnx_filepath",
+        type=str,
+        help="The full filepath of the ONNX model file or SparseZoo stub to the model",
+    )
+    parser.add_argument(
+        "-d",
+        "--dataset",
+        type=str,
```

Comment:
If dataset is not specified we get the following error:
```
Traceback (most recent call last):
  File "/home/rahul/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 274, in <module>
    main()
  File "/home/rahul/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 257, in main
    args.dataset = args.dataset.lower()
AttributeError: 'NoneType' object has no attribute 'lower'

Process finished with exit code 1
```

Suggested FIx:
```suggestion
        type=str,
        required=True,
```